deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py

```diff
@@ -0,0 +1,663 @@
+                    }
+        return (attributes)
+    def formatMessage(self, message):
+        if self.agent_version == "3.12":
+            return ('{0}:[{1}] ({2}) any->syscheck:{3}'.format(self.SYSCHECK_MQ, self.agent_id, self.agent_name, message))
+        if self.agent_version == "3.11":
+            # If first time generating. Send control message to simulate end of FIM baseline.
+            if self.baseline_completed == 0:
+                self.baseline_completed = 1
+                return ('{0}:{1}:{2}'.format(self.SYSCHECK_MQ, self.SYSCHECK, "syscheck-db-completed"))
+            return ('{0}:{1}:{2}'.format(self.SYSCHECK_MQ, self.SYSCHECK, message))
+    def generateMessage(self):
+        if self.agent_version == "3.12":
+            if self.event_type == "added":
+                timestamp = int(time())
+                self.generateAttributes()
+                attributes = self.getAttributes()
+                data = {"path": self._file, "mode": self.event_mode, "type": self.event_type, "timestamp": timestamp, "attributes": attributes}
+            if self.event_type == "modified":
+                timestamp = int(time())
+                self.generateAttributes()
+                attributes = self.getAttributes()
+                self.generateAttributes()
+                old_attributes = self.getAttributes()
+                changed_attributes = self.checkChangedAttributes(attributes, old_attributes)
+                data = {"path": self._file, "mode": self.event_mode, "type": self.event_type, "timestamp": timestamp, "attributes": attributes, "old_attributes": old_attributes, "changed_attributes": changed_attributes}
+
+            if self.event_type == "deleted":
+                timestamp = int(time())
+                self.generateAttributes()
+                attributes = self.getAttributes()
+                data = {"path": self._file, "mode": self.event_mode, "type": self.event_type, "timestamp": timestamp, "attributes": attributes}
+
+            message = json.dumps({"type": "event", "data": data})
+
+        if self.agent_version == "3.11":
+            self.generateAttributes()
+            message = '{0}:{1}:{2}:{3}:{4}:{5}:{6}:{7}:{8}:{9} {10}'.format( \
+                   self._size, self._mode, self._uid, self._gid, self._md5, \
+                   self._sha1, self._uname, self._gname, self._mdate, \
+                   self._inode, self._file)
+
+        return self.formatMessage(message)
+    def getMessage(self, event_mode=None, event_type=None):
+        if event_mode is not None:
+            self.event_mode = event_mode
+        else:
+            self.event_mode = choice(["real-time", "whodata", "scheduled"])
+
+        if event_type is not None:
+            self.event_type = event_type
+        else:
+            self.event_type = choice(["added", "modified", "deleted"])
+
+        return self.generateMessage()
```

Comment:
Add a blank line between functions and remove redundant parentheses (459, 466, 474, 475). Line 498, it is possible that the variable data is not declared if no previous if is satisfied. If the only available states are 'added', 'modified' and 'deleted', change these ifs to elfi and the last one to be an else